### PR TITLE
[IMP] website_sale : selected variants are open from cart

### PR DESCRIPTION
**Pull request by sys-odoo**
Task : https://www.odoo.com/web#id=12302&view_type=form&model=project.task&menu_id=3942&action=327
Pad : https://pad.odoo.com/p/r.D5fuz8XPmZyrXlSJ

### CODE REVIEW

**Comment by jholze** [comment]
:+1:

File: addons/website_sale/controllers/main.py [hunk 1]
```diff
@@ -236,13 +236,16 @@ def shop(self, page=0, category=None, search='', **post):
         return request.website.render("website_sale.products", values)
 
     @http.route(['/shop/product/<model("product.template"):product>'], type='http', auth="public", website=True)
-    def product(self, product, category='', search='', **kwargs):
+    def product(self, product, category='', search='', selected_product='', **kwargs):
```
Comment (hmo-odoo) [inline addons/website_sale/controllers/main.py]: rename "selected_product" with variant_id

File: addons/website_sale/controllers/main.py [hunk 2]
```diff
@@ -276,7 +279,8 @@ def product(self, product, category='', search='', **kwargs):
             'categories': categs,
             'main_object': product,
             'product': product,
-            'get_attribute_value_ids': self.get_attribute_value_ids
+            'get_attribute_value_ids': self.get_attribute_value_ids,
+            'selected_variant': selected_product if selected_product else product.product_variant_ids[0]
```
Comment (hmo-odoo) [inline addons/website_sale/controllers/main.py]: rename "selected_variant" with "variant"

Comment (sus-odoo) [inline addons/website_sale/controllers/main.py]: Done

File: addons/website_sale/controllers/main.py (hunk 1 above)
Comment (sus-odoo) [inline addons/website_sale/controllers/main.py]: Done

File: addons/website_sale/controllers/main.py [hunk 3]
```diff
@@ -245,13 +245,16 @@ def shop(self, page=0, category=None, search='', **post):
...
-    def product(self, product, category='', search='', **kwargs):
+    def product(self, product, category='', search='', variant_id='', **kwargs):
         cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
         category_obj = pool['product.public.category']
         template_obj = pool['product.template']
 
         context.update(active_id=product.id)
 
+        if variant_id:
+            variant_id = request.env['product.product'].browse(int(variant_id))
```
Comment (vme-odoo) [inline addons/website_sale/controllers/main.py]: change variable name variant_id to variant_obj

**Pull request closed without merging**